src/libstore/references.cc

```diff
@@ -54,7 +53,7 @@ void RefScanSink::operator () (std::string_view data)
        fragment, so search in the concatenation of the tail of the
        previous fragment and the start of the current fragment. */
     auto s = tail;
-    s.append(data.data(), refLength);
+    s.append(data.data(), std::min(data.size(), refLength));
```

Comment:
Slightly off topic: 
I was just debugging also trying to do `std::cout << data` to understand the codebase.
is this binary data because it wasn't printing nicely at all, it was getting converted to some control characters.

Reply:
I’m not 100% sure, but `data` here is probably a Nar archive (~ a Tar archive, but with less metadata for reproducibility). So yes, at least partially binary data

Reply:
Yes, this is a NAR.